lib/Doctrine/DBAL/Driver/IBMDB2/DB2Connection.php

```diff
@@ -126,23 +121,35 @@ public function exec(string $statement) : int
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId($name = null)
+    public function lastInsertId(?string $name = null) : string
     {
-        return db2_last_insert_id($this->conn);
+        if ($name !== null) {
+            // db2_last_insert_id() ignores the name parameter, and returns the last insert ID even if a sequence name
+            // is given. We expect an exception in that case.
+            throw new DB2Exception('This DB2 driver does not support sequences.');
```

Comment:
It's still confusing. Drivers usually don't need support for sequences, they are accessed via SQL. "Sequences on IBM DB2 are not currently supported" would be more accurate.